specs/unmanage_cluster.adoc

```diff
@@ -0,0 +1,299 @@
+    - tendrl.objects.Cluster.atoms.DeleteClusterDetails
+    - tendrl.objects.Cluster.atoms.DeleteMonitoringDetails
+  help: "Unmanage a Gluster Cluster"
+  enabled: true
+  inputs:
+    mandatory:
+      - TendrlContext.integration_id
+  run: tendrl.flows.UnmanageCluster
+  type: Update
+  uuid: 2f94a48a-05d7-408c-b400-e27827f4efed
+  version: 1
+```
+
+=== Alternatives
+
+None
+
+=== Data model impact
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* Introduce an API `cluster/{int-id}/unmanage` for triggering an un-manage
+cluster flow
+
+==== Notifications/Monitoring impact:
+
+* A flow to archive the cluster specific graphite data
+
+* A flow to remove the grafana alerts dashboards for the cluster and its
+dependent entities
+
+* Raise an alert once cluster got un-managed with details like where to look
+for old graphite data etc
+
+==== Tendrl/common impact:
+
+* A flow un-manage cluster to be targeted at tendrl server node
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl Dashboard impact:
+
+* Following changes required in UI dashboards based on UX designs mentioned at
+https://redhat.invisionapp.com/share/8QCOEVEY9
+
+** Add an option namely `Unmanage` under kebab menu for each successfully
+imported and managed cluster
+
+** Add a dialog box which opens up on click event of `Unmanage` option from
+kebab menu of the cluster. This dialog box is for confirmation from user to
+start un-manage flow for the cluster
+
+===== Workflow
+
+* User clicks the `Unmanage` option from the kebab menu for a managed cluster
+
+* The click event triggers a dialog box with appropriate message. A sample
+message is available at
+https://redhat.invisionapp.com/share/8QCOEVEY9#/screens/273239640
+
+* There are 3 possible actions on this dialog
+
+** `Close` icon to close the dialog and no action performed for un-managing the
+cluster. User would be directed back to clusters list page
+
+** `Cancel` button to close the dialog and no action performed for un-managing the
+cluster. User would be directed back to clusters list page
+
+** `Unmanage` button to start the un-manage cluster task in backend. A message
+with task details gets displayed on dialog box. Sample message available at
+https://redhat.invisionapp.com/share/8QCOEVEY9#/screens/273239844
+
+** This final message after submission of the task for un-managing cluster would
+also provide a button to view the task details. A button `View Task Progress` is
+available for the same. User can opt to close this dialog and later user context
+menus to check the task updates
+
+** Once a cluster is being moved to un-managed state, the changes in properties
+listed for cluster are as below
+
+*** `Import Status` changed to `Unmanaging`
+
+*** `Is Managed` changed to `no`
+
+*** The columns `Volume Profiling`, `Volumes` and `Alerts` would be hidden
+
+*** `View Details` link would be available to check the task details
```

Comment:
@a2batic `View Details` link is used for showing error list. The link should be named as `Track Progress` similar to Import task detail page which will take user to unmanage task detail page.  @julienlim @mcarrano  Where the unmanage task detail page will be shown in UI? Will it be similar to import task detail view  or do we need to create a global task list view to display all the global tasks.

@a2batic @shtripat Will the cluster list API response provides unmanage task_id ? What will be the value of is_managed property after triggering Unmanage action and before the unmanage task gets completed? @a2batic Please add the API details too for eg. - API url, request data, response.

Reply:
@gnehapk, yes, it should be `Task Progress`.
@gnehapk, the value for is_managed should be 'no' after triggering unmanage action and before the unmanage task gets completed. I will add the API details.

Reply:
@gnehapk @a2batic  
@mcarrano 

View Details is used to show error list, it is also used to show import task details (as it's importing).  See https://redhat.invisionapp.com/share/8QCOEVEY9#/screens/244738628.

For the unmanaged task details, it will be similar to the import task details view.  Reason is there is no active cluster anymore during an unmanage cluster.